### DISCUSSION TITLE
After the connection is established, it takes about 5 seconds before Midi packets arrive

### DISCUSSION THREAD

**Discussion by WernerWeit** [lathoub/Arduino-AppleMIDI-Library#166] [Q&A]
Hello,
I have developed a project with an ESP32 where I use RTPMidi over WLAN. But after the connection is established, it takes about 5 seconds before Midi packets arrive. I have no idea why that is. Otherwise everything works fine. Does anyone have an idea what could be causing this? Thank you

**Comment by lathoub** [comment]
I know that during the initial handshake, no messages are send.
Have you tried WireShark to see how long the handshake lasts?

**Reply by WernerWeit** [reply]
Thanks for the reply. I've never used Wireshark before but I'll give it a try. thanks for the tip

**Comment by lathoub** [comment]
Good luck - use `rtpmidi || applemidi` as filter

**Reply by WernerWeit** [reply]
i run a test connect and used wireshark but i am not shure what i see on the protocoll. what i do not understand is that i use the port 5004 but i see also traffic with port 5005. it takes about 5 seconds till the firt midi ping get throw. 
<img width="1672" alt="Bildschirmfoto 2024-03-18 um 12 24 35" src="https://github.com/lathoub/Arduino-AppleMIDI-Library/assets/79323315/54989142-d96e-4940-a21a-8bd5a69e3ef8">
192.168.1.100 is the source and 192.168.1.51 the the destination. i used a test ping on midi ch16 every second. the first message that i an see on the midi monitor ist the message wit the time 12:10:25.293 (blue).
any ideas what to read out of that? thanks. best, werner

**Reply by lathoub** [reply]
All looks OK. 

Indeed, port 5004 and 5004+1 (=5005)  are used by the protocol (AppleMIDI kinda extends RTPMidi and uses the 5004 + 1 port as well).

My guess is that on MacOS, the initiator (MIDI Network as part of the MIDI Studio on MacOS) awaits the Synchronization handshakes to stabilize.

I believe this is a bit faster on Windows (using [Tobias Ericsson's RTP MIDI](https://www.tobias-erichsen.de/software/rtpmidi.html)) and does not await the handshakes to stabilize.

No sure if what I say is correct (I need to check), but from memory that is what I have seen as well.

**Reply by WernerWeit** [reply]
Thanks for the port explanation. In practice it's actually not a big problem if you know it. I was just surprised because the connection normally works so well.
